youtube_dl/postprocessor/ffmpeg.py

```diff
@@ -307,9 +307,16 @@ def run(self, information):
             extension = self._preferredcodec
             more_opts = []
             if self._preferredquality is not None:
-                # The opus codec doesn't support the -aq option
-                if int(self._preferredquality) < 10 and extension != 'opus':
-                    more_opts += ['-q:a', self._preferredquality]
+                if int(self._preferredquality) < 10:
+                    if extension == 'opus':
```

Comment:
It's certainly possible but I'm too lazy to do that :-P

Reply:
it's not just a matter of checking extension != mp3... each codec requires it's own specific bitrate scale factor

Reply:
Luckily the other work has already been done (see the link in https://github.com/ytdl-org/youtube-dl/issues/28389#issuecomment-1129221519). Your proposal can be adapted into the `_quality_args()` method from yt-dlp, 
```py
    def _quality_args(self, codec):
        if self._preferredquality is None:
            return []
        elif self._preferredquality > 10:
            return ['-b:a', self._preferredquality + 'k']

        limits = {
            'libmp3lame': (10, 0),
            'libvorbis': (0, 10),
            # FFmpeg's AAC encoder does not have an upper limit for the value of -q:a.
            # Experimentally, with values over 4, bitrate changes were minimal or non-existent
            'aac': (0.1, 4),
            'libfdk_aac': (1, 5),
            'lib_opus': (.33, 10),
        }.get(codec)
        if not limits:
            return []
        q = limits[1] + (limits[0] - limits[1]) * (self._preferredquality / 10)
        if codec == 'libfdk_aac':
            return ['-vbr', compat_str(int(q))]

        elif codec == 'lib_opus':
            # The opus codec doesn't support the -q:a option
            # Give quality numbers 0-10 *some* meaning (as suggested by "knarrff" on Github)
            # 0 (best quality) gives 192 kb/s
            # 5 (default) gives 99 kb/s
            # 10 (worst quality) gives 6 kb/s, which is the opus minimum 
            return ['-b:a', '%.0fk' % (19.2 * q, )]
return ['-q:a', compat_str(q)]
```
This gives 99k instead of 96k for the default and 192k for the best; according to the HydrogenAudio page linked by @knarff anything over 160k is going to be better than almost all the original sources that yt-dl sees. In the attachment I've made what I hope are all the appropriate changes to include the above function.

[30957qa.patch](https://github.com/ytdl-org/youtube-dl/files/8794741/30957qa.patch.txt)

If a mapping that can't use the `x + (y-x)*q/10` formula is really critical, the calculation and return could go before the definition of `limits`.